libraries/Microsoft.Bot.Builder.AI.LUIS/LuisV3Recognizer.cs

```diff
@@ -0,0 +1,510 @@
+        {
+            _application = application ?? throw new ArgumentNullException(nameof(application));
+            _options = predictionOptions ?? new LuisV3PredictionOptions();
+            _includeApiResults = includeApiResults;
+
+            TelemetryClient = _options.TelemetryClient;
+            LogPersonalInformation = _options.LogPersonalInformation;
+
+            var delegatingHandler = new LuisDelegatingHandler();
+            var httpClientHandler = clientHandler ?? CreateRootHandler();
+            var currentHandler = CreateHttpHandlerPipeline(httpClientHandler, delegatingHandler);
+
+            DefaultHttpClient = new HttpClient(currentHandler, false)
+            {
+                Timeout = TimeSpan.FromMilliseconds(_options.Timeout),
+            };
+
+            DefaultHttpClient.DefaultRequestHeaders.Add("Ocp-Apim-Subscription-Key", _application.EndpointKey);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LuisV3Recognizer"/> class.
+        /// </summary>
+        /// <param name="service">The LUIS service from configuration.</param>
+        /// <param name="predictionOptions">(Optional) The LUIS prediction options to use.</param>
+        /// <param name="includeApiResults">(Optional) TRUE to include raw LUIS API response.</param>
+        /// <param name="clientHandler">(Optional) Custom handler for LUIS API calls to allow mocking.</param>
+        public LuisV3Recognizer(LuisService service, LuisV3PredictionOptions predictionOptions = null, bool includeApiResults = false, HttpClientHandler clientHandler = null)
+            : this(new LuisApplication(service), predictionOptions, includeApiResults, clientHandler)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LuisV3Recognizer"/> class.
+        /// </summary>
+        /// <param name="applicationEndpoint">The LUIS endpoint as shown in https://luis.ai .</param>
+        /// <param name="predictionOptions">(Optional) The LUIS prediction options to use.</param>
+        /// <param name="includeApiResults">(Optional) TRUE to include raw LUIS API response.</param>
+        /// <param name="clientHandler">(Optional) Custom handler for LUIS API calls to allow mocking.</param>
+        public LuisV3Recognizer(string applicationEndpoint, LuisV3PredictionOptions predictionOptions = null, bool includeApiResults = false, HttpClientHandler clientHandler = null)
+            : this(new LuisApplication(applicationEndpoint), predictionOptions, includeApiResults, clientHandler)
+        {
+        }
+
+        /// <summary>
+        /// Gets client to use for http.
+        /// </summary>
+        /// <value>
+        /// Client to use for http.
+        /// </value>
+        public static HttpClient DefaultHttpClient { get; private set; }
+
```

Comment:
See comment around the IHttpClientFactory. We also have a ConstantHttpClientFactory you can re-use here.